bin/citrea/tests/evm/uniswap/scripts/01_deploy.js

```diff
@@ -1,109 +1,149 @@
-
-    const approvalUsdtOwnerA = await usdt.connect(owner).approve(router.address, constants.MaxUint256);
-    await approvalUsdtOwnerA.wait();
-    const approvalUsdcOwnerA = await usdc.connect(owner).approve(router.address, constants.MaxUint256);
-    await approvalUsdcOwnerA.wait();
-    const approvalUsdtTraderA = await usdt.connect(trader).approve(router.address, constants.MaxUint256);
-    await approvalUsdtTraderA.wait();
-    const approvalUsdcTraderA = await usdc.connect(trader).approve(router.address, constants.MaxUint256);
-    await approvalUsdcTraderA.wait();
-    
-
-    const addLiquidityTx = await router.connect(owner).addLiquidity(
-        usdt.address,
-        usdc.address,
-        utils.parseEther('100'),
-        utils.parseEther('100'),
-        0,
-        0,
-        owner.address,
-        Math.floor(Date.now() / 1000 + (10 * 60)),
-        { gasLimit: utils.hexlify(1_000_000)}
+  console.log("Starting Uniswap deployment script...");
+  const [owner, trader] = await ethers.getSigners();
+  // assert that owner.address and trader.address are not undefined
+  if (!owner || !owner.address || !trader || !trader.address) {
+    throw new Error("Could not get owner and trader addresses");
+  }
+
+  console.log(
+    `Starting with owner=${owner.address} and trader=${trader.address}`,
+  );
+
+  console.log("Deploying USDT...");
+  const Usdt = await ethers.getContractFactory("contracts/Tether.sol:Tether", owner);
+  const usdt = await Usdt.deploy({nonce: 0});
+  console.log("USDT deployed at:", usdt.address);
+  console.log("Deploying USDC...");
+  const Usdc = await ethers.getContractFactory("contracts/UsdCoin.sol:UsdCoin", owner);
+  const usdc = await Usdc.deploy({nonce: 1});
+  console.log("USDC deployed at:", usdc.address);
+  console.log("Deploying WETH...");
+  const Weth = new ContractFactory(WETH9.abi, WETH9.bytecode, owner);
+  const weth = await Weth.deploy({nonce: 2});
+  console.log("WETH deployed at:", weth.address);
+
+  console.log("Minting some tokens...");
+  const mintAmount = utils.parseEther("100000");
+  await usdt.connect(owner).mint(owner.address, mintAmount, {nonce: 3});
+  console.log("Minted USDT for owner");
+  await usdc.connect(owner).mint(owner.address, mintAmount, {nonce: 4});
+  console.log("Minted USDC for owner");
+  await usdt.connect(owner).mint(trader.address, mintAmount, {nonce: 5});
+  console.log("Minted USDT for trader");
+  await usdc.connect(owner).mint(trader.address, mintAmount, {nonce: 6});
+  console.log("Minted USDC for trader");
+  console.log("All tokens have been minted");
+
+  console.log("Deploying Uniswap factory...");
+  const Factory = new ContractFactory(
+    factoryArtifact.abi,
+    factoryArtifact.bytecode,
+    owner,
+  );
+  const factory = await Factory.deploy(owner.address, { nonce: 7 });
+  console.log("Factory deployed at:", factory.address);
+
+  console.log("Creating USDT/USDC pair...");
+  const tx = await factory.createPair(usdt.address, usdc.address);
```

Comment:
looks like you should supply nonce here as well?

Reply:
It works without an extra nonce